src/elements/Button/Button.js

```diff
@@ -1,34 +1,136 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
+
 import META from '../../utils/Meta'
+import * as sui from '../../utils/semanticUtils'
+import {
+  iconPropRenderer,
+  useKeyOrValueAndKey,
+  getUnhandledProps,
+  useValueAndKey,
+  useKeyOnly,
+} from '../../utils/propUtils'
+import ButtonGroup from './ButtonGroup'
+
+/**
+ * Button
+ */
+function Button(props) {
+  const {
+    children, color, size, className, animated, floated, labeled, icon, inverted, fluid, basic, active, disabled,
+    loading, circular, compact, attached, toggle, type,
+  } = props
+
+  const classes = cx('sd-button', 'ui',
+    icon && 'icon',
```

Comment:
We have a tricky bit here to deal with.  The presence of an icon will not always necessitate the icon class.  The icon class collapses padding to work with individual icon buttons:

![](http://g.recordit.co/NxLAURIXRj.gif)

I'm not sure of a good solution here yet.  I'll go through this the examples and your suggestion about `Button.Left`, etc in more detail later.